api/src/main/java/com/datatorrent/api/Context.java

```diff
@@ -301,6 +301,12 @@
      */
     int getId();
 
+    /**
+     * Return the number of windows before the next checkpoint including the current window.
+     * @return Number of windows from checkpoint, 1 if the checkpoint will be after the current window
+     */
+    int getWindowsFromCheckpoint();
```

Comment:
@chandnisingh semver didn't flag it. In theory it is incompatible, in practice it isn't since users use this interface as opposed to implementing it today. Given that this is in API and there is no such distinction advertised today I will consider moving it to something like an attribute.